snapcraft/parts/plugins/matter_plugin.py

```diff
@@ -0,0 +1,138 @@
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = plugins.extract_plugin_properties(
+            data,
+            plugin_name="matter",
+            # required=["matter_branch", "zap_version"]
+        )
+        return cls(**plugin_data)
+
+
+class MatterPlugin(plugins.Plugin):
+    """A plugin for matter project.
+
+    This plugin uses the common plugin keywords.
+    For more information check the 'plugins' topic.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+        - matter-branch
+          (str, no default)
+          The matter branch to use for the build.
+        - zap-version
+          (str, no default)
+          The zap version to use for the build.
+    """
+
+    properties_class = MatterPluginProperties
+
+    def __init__(
+        self,
+        *,
+        properties: plugins.PluginProperties,
+        part_info: infos.PartInfo,
+    ) -> None:
+        super().__init__(properties=properties, part_info=part_info)
+
+        self.matter_dir = part_info.part_build_dir
+        self.snap_arch = os.getenv("SNAP_ARCH")
+
+    @overrides
+    def get_build_packages(self) -> Set[str]:
+        return {
+            "wget",
+            "unzip",
+            "clang",
+            "pkg-config",
+            "git",
+            "cmake",
+            "ninja-build",
+            "unzip",
+            "libssl-dev",
+            "libdbus-1-dev",
+            "libglib2.0-dev",
+            "libavahi-client-dev",
+            "python3-venv",
+            "python3-dev",
+            "python3-pip",
+            "libgirepository1.0-dev",
+            "libcairo2-dev",
+            "libreadline-dev",
+            "generate-ninja",
+        }
+
+    @overrides
+    def get_build_environment(self) -> Dict[str, str]:
+        return {}
+
+    @overrides
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    @overrides
+    def get_build_commands(self) -> List[str]:
+        commands = []
+
+        if self.snap_arch == "arm64":
+            commands.extend(
+                [
+                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/v2023.11.13/zap-linux-{self.snap_arch}.zip",
+                    f"unzip -o zap-linux-{self.snap_arch}.zip",
+                    "echo 'export ZAP_INSTALL_PATH=$PWD'",
+                ]
+            )
+
+        commands.extend(
+            [
+                f"if [ ! -d matter ]; then git clone --depth 1 -b v1.2.0.1 {MATTER_REPO} matter; fi",
+                "cd matter || echo 'skip clone'",
+                f"scripts/checkout_submodules.py --shallow --platform linux",
+                f"set +u && source scripts/activate.sh && set -u",
+                "cp -vr ./* $CRAFT_PART_INSTALL/",
+                "echo 'Cloned Matter repository and activated submodules'",
+            ]
+        )
+
+        return commands
```

Comment:
this matter plugin doesn't really build anything, does it? it might be more suitable as an extension

Reply:
Sourcing `scripts/activate.sh` (line 132) builds the SDK and sets up the environment in preparation for the application build. In the recent versions, this is actually just a [symbolic link](https://github.com/project-chip/connectedhomeip/blob/v1.2.0.1/scripts/activate.sh) to [scripts/setup/bootstrap.sh](https://github.com/project-chip/connectedhomeip/blob/v1.2.0.1/scripts/setup/bootstrap.sh), kept there for backward compatibility. I suggest replacing it with the actual target, which has a clearer name.

Reply:
This Matter plugin builds the Matter SDK. Some comments have been added for better clarity.